pkg/sql/sqlbase/type_desc.go

```diff
@@ -188,13 +208,33 @@ func (desc *MutableTypeDescriptor) MaybeIncrementVersion() {
 	desc.ModificationTime = hlc.Timestamp{}
 }
 
+// OriginalName implements the MutableDescriptor interface.
+func (desc *MutableTypeDescriptor) OriginalName() string {
+	return desc.ClusterVersion.Name
```

Comment:
The `clusterVersion` on a `MutableTypeDescriptor` is a pointer, so it'll be nil when the descriptor is new, so this might panic. Should you change it here to not be a pointer?

Reply:
Good point. The same is true for databases and schemas, and I think we should keep the `ClusterVersion` as pointers since these are immutable descriptors. I added nil checks for all of these fields.

In the future, though, I'd like to get rid of `ClusterVersion` since there's no good reason to keep the whole descriptor around. I think we really only need a few pieces of information (name, version, whether the descriptor is new, and maybe `NextMutationID`).